solidity/test/KeepFactorySelectionTest.js

```diff
@@ -212,27 +360,193 @@ describe("KeepFactorySelection", async () => {
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(newValue)
+    })
+
+    // KEEP stake vendor set, factory set.
+    // ETH-only vendor set, factory not set.
+    it("updates value in locked factories", async () => {
+      await keepFactorySelection.setFullyBackedKeepVendor(
+        fullyBackedVendor.address,
+      )
+
+      // Lock factories
+      await keepFactorySelection.lockFactoriesVersions(
+        keepStakedFactory.address,
+        fullyBackedFactory.address,
+      )
+
+      // Upgrade factories in vendors
+      const newKeepFactory = await ECDSAKeepFactoryStub.new()
+      await keepStakedVendor.setFactory(newKeepFactory.address)
+
+      const newEthFactory = await ECDSAKeepFactoryStub.new()
+      await fullyBackedVendor.setFactory(newEthFactory.address)
+
+      // Set minimum bondable value
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(newValue)
+
+      expect(await newKeepFactory.minimumBondableValue()).to.eq.BN(defaultValue)
+      expect(await newEthFactory.minimumBondableValue()).to.eq.BN(defaultValue)
+    })
   })
 
-  describe("setFullyBackedKeepFactory", async () => {
+  describe("setFullyBackedKeepVendor", async () => {
     it("can be called only one time", async () => {
-      await keepFactorySelection.setFullyBackedKeepFactory(
-        ethStakeFactory.address,
+      await keepFactorySelection.setFullyBackedKeepVendor(
+        fullyBackedVendor.address,
       )
       // ok, this was the first time
 
       await expectRevert(
-        keepFactorySelection.setFullyBackedKeepFactory(ethStakeFactory.address),
-        "Fully backed factory already set",
+        keepFactorySelection.setFullyBackedKeepVendor(
+          fullyBackedVendor.address,
+        ),
+        "Fully backed vendor already set",
       )
     })
 
     it("can not be called for 0 address", async () => {
       await expectRevert(
-        keepFactorySelection.setFullyBackedKeepFactory(constants.ZERO_ADDRESS),
+        keepFactorySelection.setFullyBackedKeepVendor(constants.ZERO_ADDRESS),
         "Invalid address",
       )
     })
+
+    it("obtains factory from vendor", async () => {
+      await keepFactorySelection.setFullyBackedKeepVendor(
+        fullyBackedVendor.address,
+      )
+
+      await keepFactorySelection.setKeepFactorySelector(
+        keepFactorySelector.address,
+      )
+
+      await keepFactorySelector.setFullyBackedMode()
+
+      // refresh the choice
+      await keepFactorySelection.selectFactoryAndRefresh()
+
+      expect(
+        await keepFactorySelection.selectFactory(),
+        "unexpected factory selected",
+      ).to.equal(fullyBackedFactory.address)
+    })
```

Comment:
Does this case belong to `describe("setFullyBackedKeepVendor"` ? I think this behavior is pretty well covered before in all the selection tests we are having.